### PR TITLE
Add SynBN.

### PR DESCRIPTION
**Pull request by sufeidechabei**

### CODE REVIEW

File: gluoncv/model_zoo/mobilenet.py [hunk 1]
```diff
@@ -43,17 +54,23 @@ def hybrid_forward(self, F, x):
 
 # pylint: disable= too-many-arguments
 def _add_conv(out, channels=1, kernel=1, stride=1, pad=0,
-              num_group=1, active=True, relu6=False):
-    out.add(nn.Conv2D(channels, kernel, stride, pad, groups=num_group, use_bias=False))
-    out.add(nn.BatchNorm(scale=True))
+              num_group=1, active=True, relu6=False, num_sync_bn_devices=-1):
+    out.add(nn.Conv2D(channels,kernel,stride,pad,groups=num_group,use_bias=False))
+    
+    if num_sync_bn_devices == -1:
```
Comment (zhreshold) [inline gluoncv/model_zoo/mobilenet.py]: if num_sync_bn_devices < 1

Comment (sufeidechabei) [reply gluoncv/model_zoo/mobilenet.py]: okay, I have changed it

File: gluoncv/model_zoo/mobilenet.py [hunk 2]
```diff
@@ -43,17 +54,23 @@ def hybrid_forward(self, F, x):
 
 # pylint: disable= too-many-arguments
 def _add_conv(out, channels=1, kernel=1, stride=1, pad=0,
-              num_group=1, active=True, relu6=False):
-    out.add(nn.Conv2D(channels, kernel, stride, pad, groups=num_group, use_bias=False))
-    out.add(nn.BatchNorm(scale=True))
+              num_group=1, active=True, relu6=False, num_sync_bn_devices=-1):
+    out.add(nn.Conv2D(channels,kernel,stride,pad,groups=num_group,use_bias=False))
```
[the remark this answers is not in the document]
Comment (sufeidechabei) [reply gluoncv/model_zoo/mobilenet.py]: Okay, I have fixed it.

File: gluoncv/model_zoo/mobilenet.py [hunk 3]
```diff
@@ -204,17 +229,17 @@ def get_mobilenet(multiplier, pretrained=False, ctx=cpu(),
...
         Location for keeping the model parameters.
+    num_sync_bn_devices : int, default is -1
+        Number of devices for training. If `num_sync_bn_devices < 2`, SyncBatchNorm is disabled.
     """
-    net = MobileNet(multiplier, **kwargs)
-
+    net = MobileNet(multiplier, num_sync_bn_devices=num_sync_bn_devices, **kwargs)
     if pretrained:
-        from .model_store import get_model_file
+        from gluoncv.model_zoo.model_store import get_model_file
```
Comment (zhreshold) [inline gluoncv/model_zoo/mobilenet.py]: from inside the gluoncv, import gluoncv is not allowed

File: gluoncv/model_zoo/mobilenet.py [hunk 4]
```diff
@@ -204,17 +229,17 @@ def get_mobilenet(multiplier, pretrained=False, ctx=cpu(),
...
         version_suffix = '{0:.2f}'.format(multiplier)
         if version_suffix in ('1.00', '0.50'):
             version_suffix = version_suffix[:-1]
-        net.load_parameters(
-            get_model_file('mobilenet%s' % version_suffix, tag=pretrained, root=root), ctx=ctx)
-        from ..data import ImageNet1kAttr
+        net.load_parameters(get_model_file('mobilenet%s' % version_suffix, tag=pretrained, root=root), ctx=ctx)
+        from gluoncv.data import ImageNet1kAttr
```
Comment (zhreshold) [inline gluoncv/model_zoo/mobilenet.py]: same here, import is not valid

File: gluoncv/model_zoo/mobilenet.py [hunk 5]
```diff
@@ -43,17 +54,23 @@ def hybrid_forward(self, F, x):
...
-              num_group=1, active=True, relu6=False):
+              num_group=1, active=True, relu6=False, num_sync_bn_devices=-1):
     out.add(nn.Conv2D(channels, kernel, stride, pad, groups=num_group, use_bias=False))
-    out.add(nn.BatchNorm(scale=True))
+    
+    if num_sync_bn_devices <= 1:
+        out.add(nn.BatchNorm(scale=True))
+    else:
+        out.add(gluon.contrib.nn.SyncBatchNorm(
+            epsilon=1e-5, momentum=0.9, num_devices=num_sync_bn_devices))
```
Comment (zhreshold) [inline gluoncv/model_zoo/mobilenet.py]: please match the default arguments with the non sync version, nn.BatchNorm(scale=True) and SyncBatchNorm(scale=True, num_devices=num_sync_bn_devices)

Comment (sufeidechabei) [reply gluoncv/model_zoo/mobilenet.py]: Okay, I have fixed it.

**Comment by zhreshold** [comment]
LGTM, please fix lint error, and please add a unittest similar in https://github.com/dmlc/gluon-cv/blob/master/tests/unittests/test_model_zoo.py

For example, a test function `test_mobilenet_sync_bn()` which tests the weights from mobilenet can be correctly loaded by sync bn version.

File: tests/unittests/test_model_zoo.py [hunk 6]
```diff
@@ -193,7 +193,15 @@ def test_segmentation_models():
...
+    _test_model_list(models, ctx, x)
```
Comment (zhreshold) [inline tests/unittests/test_model_zoo.py]: no, it's testing normal version inference, I mean a test to assert that the weights from normal version can be loaded to sync bn version.
Our test CI currently only use 1 GPU, so there's no way to test inference of sync bn mobilenet. We will add that later.

**Comment by zhreshold** [comment]
Thanks for the contribution, this is merged.
